globus_automate_client/cli/rich_helpers.py

```diff
@@ -0,0 +1,435 @@
+    authzs = get_authorizers_for_scopes(["openid"], no_login=True)
+    authorizer = authzs.get("openid")
+    if authorizer is None:
+        return l
+
+    # Collect IDs from the List data
+    creators: Dict[str, None] = collections.OrderedDict()
+    for item in l:
+        urn_id = item.get(field, "")
+        id = urn_id.split(":")[-1]
+        creators[id] = None
+
+    # Get id -> username mapping
+    ac = AuthClient(authorizer=authorizer)
+    resp = ac.get_identities(ids=creators.keys())
+    id_to_user = {i["id"]: i["username"] for i in resp.data["identities"]}
+
+    # Update items in list
+    for item in l:
+        urn_id = item.get(field, "")
+        id = urn_id.split(":")[-1]
+        if id in id_to_user:
+            item[field] = id_to_user[id]
+    return l
+
+
+class Field:
+    """
+    A generic class structure for transforming lists of data into a table. Each
+    instance of this class represents a single field to pull from a data item.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        default: str,
+        transformation: Optional[Callable[[str], str]] = None,
+    ) -> None:
+        """
+        name: The name of the field which contains the data for display
+        default: A placeholder to use if the field is not available
+        transformation: A callable that takes and returns a string. This is used
+            to format the data field in an item
+        """
+        self.name = name
+        self.default = default
+        self.transformation = transformation
+
+
+class DisplayFields(abc.ABC):
+    """
+    An object representing how to parse lists of data into a table.
+
+    fields: A list of Fields that defines which fields get rendered into a table
+        and how they are presented
+    path_to_data_list: A key under which the data list is defined
+    """
+
+    fields: List[Field]
+    path_to_data_list: str
+    prehook: Optional[Callable[[List[Dict[str, Any]]], List[Dict[str, Any]]]] = None
+
+
+class RunListDisplayFields(DisplayFields):
+    """
+    This object defines the fields and display style of a Run listing into a
+    table.
+    """
+
+    fields = [
+        Field("action_id", ""),
+        Field("label", "<EMPTY>"),
+        Field("status", ""),
+        Field("start_time", "", humanize_datetimes),
+        Field("created_by", "", humanize_auth_urn),
+        Field("flow_id", "<DELETED>"),
+    ]
+    path_to_data_list = "actions"
+    prehook = functools.partial(identity_to_user, "created_by")
+
+
+class RunEnumerateDisplayFields(RunListDisplayFields):
+    """
+    This object defines the fields and display style of a Run enumeration into a
+    table.
+    """
+
+    path_to_data_list = "runs"
+
+
+class FlowListDisplayFields(DisplayFields):
+    """
+    This object defines the fields and display style of a Flow listing into a
+    table.
+    """
+
+    fields = [
+        Field("title", ""),
```

Comment:
adding another idea when I see this again: Could the `Field` be an Enum of some sort where these name to transformation things are properties of the field in the Enum. Thus, maybe, that Enum could also end up being the type for the Typer parameter for `--output-fields` or some such.

Again, I don't think we have to roll that into this PR, but mgiht be something to address soon after?

Reply:
Could be! I think it depends on what users want to do, ideally we'd be able to have a single generic view that works and makes everyone happy. This CLI bit of code is already more complicated than I'd originally hoped.